sympy/matrices/__init__.py

```diff
@@ -25,11 +26,28 @@
     MatrixSlice, BlockDiagMatrix, BlockMatrix, FunctionMatrix, Identity,
     Inverse, MatAdd, MatMul, MatPow, MatrixExpr, MatrixSymbol, Trace,
     Transpose, ZeroMatrix, OneMatrix, blockcut, block_collapse, matrix_symbols, Adjoint,
-    hadamard_product, HadamardProduct, HadamardPower, Determinant, det,
+    hadamard_product, HadamardProduct, HadamardPower, Determinant,
     diagonalize_vector, DiagMatrix, DiagonalMatrix, DiagonalOf, trace,
     DotProduct, kronecker_product, KroneckerProduct,
     PermutationMatrix, MatrixPermute)
 
+from .determinant import (
+    adjugate, charpoly, cofactor, cofactor_matrix,
+    det, det_bareiss, det_berkowitz, det_LU, minor, minor_submatrix)
+
+from .reductions import (
+    is_echelon, echelon_form, rank, rref)
+
+from .subspaces import (
+    columnspace, nullspace, rowspace, orthogonalize)
+
+from .eigen import (
+    eigenvals, eigenvects, is_diagonalizable, diagonalize,
+    is_positive_definite, is_positive_semidefinite, is_negative_definite,
+    is_negative_semidefinite, is_indefinite,
+    jordan_form, left_eigenvects, singular_values)
```

Comment:
The only ones I would export here are `eigenvals`, `eigenvects`, `diagonalize`, `jordan_form` and `singular_values`. I'm not sure which would make sense for matrix expressions. I can imagine wanting to have `EigenvalueOf(M, n)` for the `n`nth eigenvalue.

Reply:
So again, some clarification please, when you mean export are you saying I should remove those functions completely and from `matrices.__init__.py` or leave them there to be imported `from sympy.matrices import *` but not carry over to the top level sympy import?

Reply:
I wouldn't import e.g. `is_positive_definite` here in `sympy/matrices/__init__.py`.

Reply:
Ok, that is what I wanted to know, so same treatment for `is_echelon`, `echelon_form` and `rref` then I assume.